### PR TITLE
Refactor contacts

### PR DESCRIPTION
**Pull request by olzraiti**

### CODE REVIEW

**Comment by olzraiti** [comment]
Retest this please

**Comment by olzraiti** [comment]
Retest this please

**Comment by olzraiti** [comment]
This closes #422 and #405

**Comment by olzraiti** [comment]
Retest this please

**Comment by olzraiti** [comment]
Retest this please

**Comment by ffagerho** [comment]
Retest this please

File: App/src/main/java/cc/softwarefactory/lokki/android/fragments/MapViewFragment.java [hunk 1]
```diff
@@ -592,32 +558,17 @@ protected void onPostExecute(Bitmap bitmapResult) {
...
-                Person p = new Person(latLng, email, time, bitmapResult);
-                markerMap.put(email, p);
-                clusterManager.addItem(p);
-            } else {
-                Log.d(TAG, "onPostExecute - creating marker: " + email);
-                marker = new Person(latLng, email, time, bitmapResult);
+            Person marker = person;
+
+            if (!markers.contains(marker)) {
+                markers.add(marker);
```
Comment (akkashaider) [inline App/src/main/java/cc/softwarefactory/lokki/android/fragments/MapViewFragment.java]: You have to remove marker from clustermanager and add it again, as there is no way to update cluster

**Pull request merged**